From 38f60de83d3414c7fecb98792b98bb87f817b31e Mon Sep 17 00:00:00 2001
From: "Evgeny Grin (Karlson2k)" <k2k@narod.ru>
Date: Thu, 21 Apr 2022 09:19:44 +0300
Subject: [PATCH] test_https_sni: use certificates with SAN fields

While host in CommonName (CN) is still should be supported, it is
recommended to use SubjectAltName to identify the host.
This also workarounds the bug in libcurl 7.82.0 with certificates
processing.
---
 src/testcurl/https/host1.crt        | 30 -----------------------------
 src/testcurl/https/host1.key        | 28 ---------------------------
 src/testcurl/https/host2.crt        | 30 -----------------------------
 src/testcurl/https/host2.key        | 28 ---------------------------
 src/testcurl/https/mhdhost1.crt     | 30 +++++++++++++++++++++++++++++
 src/testcurl/https/mhdhost1.key     | 28 +++++++++++++++++++++++++++
 src/testcurl/https/mhdhost2.crt     | 30 +++++++++++++++++++++++++++++
 src/testcurl/https/mhdhost2.key     | 28 +++++++++++++++++++++++++++
 src/testcurl/https/test_https_sni.c | 14 ++++++++------
 9 files changed, 124 insertions(+), 122 deletions(-)
 delete mode 100644 src/testcurl/https/host1.crt
 delete mode 100644 src/testcurl/https/host1.key
 delete mode 100644 src/testcurl/https/host2.crt
 delete mode 100644 src/testcurl/https/host2.key
 create mode 100644 src/testcurl/https/mhdhost1.crt
 create mode 100644 src/testcurl/https/mhdhost1.key
 create mode 100644 src/testcurl/https/mhdhost2.crt
 create mode 100644 src/testcurl/https/mhdhost2.key

diff --git a/src/testcurl/https/host1.crt b/src/testcurl/https/host1.crt
deleted file mode 100644
index dfdc0969..00000000
--- a/src/testcurl/https/host1.crt
+++ /dev/null
@@ -1,30 +0,0 @@
------BEGIN CERTIFICATE-----
-MIIFDjCCAvagAwIBAgIBAjANBgkqhkiG9w0BAQsFADCBgTELMAkGA1UEBhMCUlUx
-DzANBgNVBAgMBk1vc2NvdzEPMA0GA1UEBwwGTW9zY293MRswGQYDVQQKDBJ0ZXN0
-LWxpYm1pY3JvaHR0cGQxITAfBgkqhkiG9w0BCQEWEm5vYm9keUBleGFtcGxlLm9y
-ZzEQMA4GA1UEAwwHdGVzdC1DQTAgFw0yMTA0MTAxMjM2MTZaGA8yMTIxMDMxNjEy
-MzYxNlowXDELMAkGA1UEBhMCUlUxDzANBgNVBAgMBk1vc2NvdzEPMA0GA1UEBwwG
-TW9zY293MRswGQYDVQQKDBJ0ZXN0LWxpYm1pY3JvaHR0cGQxDjAMBgNVBAMMBWhv
-c3QxMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA6yg4JZH7R2We2Wa0
-12TqU/6D/ur17TEGF2D5Mdn5neEk9fWe2coNvz9/CKgez39QaQKR5pUjtCt9QZNX
-4yenzulazMx7KhuI/rFPw0+BU1HnN+hUvmQZjbexl9E7u5QIfB2dD9UFUNhZRQHY
-I/UNd2nsVAMhKscoWcVQwNvO0tY3wYX88jrl4nKJlJdEMhKeHe0RMi6vdfZNx7Yi
-JsBMUEgj1wta9A+TLuMOqZnCktQR1nIAg7m4C3lWEZ/OWs0zGlNmwN81/A32+D4D
-ohCCLdyJsVcHmatEE8EmbjCSo9SVkOz/722UV8gvwlyhHa6c23a3t/dXjivCfK7+
-FiDskQIDAQABo4GyMIGvMAsGA1UdDwQEAwIFoDAMBgNVHRMBAf8EAjAAMBYGA1Ud
-JQEB/wQMMAoGCCsGAQUFBwMBMB0GA1UdDgQWBBQU3oKnON9eYczaV3TJKunbobf6
-uzAnBglghkgBhvhCAQ0EGhYYVGVzdCBsaWJtaWNyb2h0dHBkIGhvc3QxMBEGCWCG
-SAGG+EIBAQQEAwIGQDAfBgNVHSMEGDAWgBRYdUPApWoxw4U13Rqsjf9AHdbpLDAN
-BgkqhkiG9w0BAQsFAAOCAgEAKnzPKJ38JP30UE6bpwRy05Frnwt/0rX6B1a9a87M
-LOwjaqDQb8p1z22+xyqFYPfkvCqRuyEjyHdu7iMxmMgSlK9nT9NZVFHeM/+/E5p7
-cMngEwc/kLMwetaT1XFrb6zZ32Z61c56gprbGQhfLgvUocie4htv+ultMZtsTUx/
-YbwifcfO4aewwmq93YSxrNXIamSZdsBo/ET+eoouGAdPIeohXNxIJkjwLoN7MlKH
-9G+12N6DfJ+RJjoljfWuf9683VJzRYeVIJm8b/ZQ9sIeaVjbEM9vpwYVCX4GgfJV
-oEd+GFIj2NpLHYhnn757G7z9wmmc0KzutGhRoH8yQo5BHsxWBnxM64whCnvE6o88
-xPopbcJQrQdbrBxr7barMWUz9bPKcuSHVjya0G2ovlJ+gmXaug/FVOmNNG1Psqxe
-cwLuJYCjIk+ZElFRh3mo0UG0vJBF35wJSQZVEtJQ7sKutw6U+oU0r6yS/XsRhU4y
-p+/A23Z89haHB3phEBKZ923QIKmGofjjkXxMaGfp4FZKrGn7JmT4TQ7qjoSmJFPJ
-80a6HJbUK9rYjqdkxlMomJ2nCCcVdnzyj8apNx+cDkaIClmlUJROVuvmXJafQfml
-Pk3B3jzOn0slaqIQCkhyPi4v+7kYxG5plKhSgLwWaHx8ft6KrjXFDR920YSdOwPR
-OxY=
------END CERTIFICATE-----
diff --git a/src/testcurl/https/host1.key b/src/testcurl/https/host1.key
deleted file mode 100644
index 7217fb61..00000000
--- a/src/testcurl/https/host1.key
+++ /dev/null
@@ -1,28 +0,0 @@
------BEGIN PRIVATE KEY-----
-MIIEvgIBADANBgkqhkiG9w0BAQEFAASCBKgwggSkAgEAAoIBAQDrKDglkftHZZ7Z
-ZrTXZOpT/oP+6vXtMQYXYPkx2fmd4ST19Z7Zyg2/P38IqB7Pf1BpApHmlSO0K31B
-k1fjJ6fO6VrMzHsqG4j+sU/DT4FTUec36FS+ZBmNt7GX0Tu7lAh8HZ0P1QVQ2FlF
-Adgj9Q13aexUAyEqxyhZxVDA287S1jfBhfzyOuXicomUl0QyEp4d7REyLq919k3H
-tiImwExQSCPXC1r0D5Mu4w6pmcKS1BHWcgCDubgLeVYRn85azTMaU2bA3zX8Dfb4
-PgOiEIIt3ImxVweZq0QTwSZuMJKj1JWQ7P/vbZRXyC/CXKEdrpzbdre391eOK8J8
-rv4WIOyRAgMBAAECggEBAKAuu2BMOmB+BmCMp675XO1uN7eTkYXKmR2fvt+IXZJ/
-5M9uSpCMJCUcOYpKCwLI9JszVvM8sB9waFgc6Ach4hr9tg4/U75xFzZ2zaB7Ldr1
-tYzNw7wVL5rrgBusOQYjZkw1V+GK954EYXqyMOr9Z/jpvEPWBee/1eqgj8tvb0kJ
-uJgQUPNqPOcf72y71Z7jnHt4TZdQVwABgiPVYu+xKfK/UFB7pXhVpxQSD6n9G4O+
-7DUMw9FcOM01R55fr3lty3yNJjCwR8SGVSgqBAS9UkayaNCjjnghXG1RUdXUw8ox
-fNAh3JBnGAiv3ttmfBzpUBAcIFCF8dgrBTkLiJ98k+kCgYEA9nr1L+jKGXDn5h8C
-DaFH9Gn7Ww3vnJ2cNiKGu5NNoz09fpsWstaDVc8ePW7+wAvecFSx8Kc6rF7rxb1L
-UuoRrVfOs9G4pocvEgeAG16j2pr7CrTubmx9XRKy/blesZByerOliI5Iqo6OnZzX
-809dcxCQc5Le9GGVrokWGk1wxLMCgYEA9D1ODo+ZMNAEvvsqssVby6UfKDnjiZ3G
-1z0v0hSTt1qT/go82r1ilm7AljWFMtpNZ1As8aDVjjTLApMGDtKFhWb97whkKGoL
-GQSz4d7s2XbBP3naxw3qbUp+LomW9BvZ/B48Xy2mdvt+gJd3fgp5sG5DW5hnPwvN
-hc3cHJKa06sCgYAmAwJrGBCK6eGpmKCdb4EETY/pE/YfBiOAKcTzip0PjrM8IjjW
-7cQlaKK2QLbi8TGlyzoB7hhbWuwre6zxJY4CWKdVDgPXEsQMmMGF9wt3/Cy5Evwu
-ZWbUAr096eeTpL4y9B+UNrU4MHIIFpcrO9MimqxcVosxzxbhatzQw56fhwKBgQDB
-f/hhjN8fTw8Th8YyUC4Dz1Q5+HMOylOIn09Nr0PAliB9ufqwii6bZC9sUgavXUjJ
-uzmATZ4jxiu4ZGb+Q0fJgulYAVvosOsN7S3em1EmkPFVquLrpVi/Ma96i0NvLsmq
-9wF5np8t0N5OXrIqx33Vi+YNoVbbGiy//3KNqJDFdQKBgAFNQS3dFXSK/tKS59vY
-QvFI82x8SR/H+UTbfpGwfJomTVSGJFPT+YMrWBeiFeUCv+W6ZnniRjN3Ka5+9Gb8
-StvzBRex61NH1+K+06DthvdRMHMgD64mHx60ZUPbf1tzDC5vIgFTKjW42tSlYkdh
-uYkjXv3kGk8Wwk6I4lhcqOVu
------END PRIVATE KEY-----
diff --git a/src/testcurl/https/host2.crt b/src/testcurl/https/host2.crt
deleted file mode 100644
index 15670618..00000000
--- a/src/testcurl/https/host2.crt
+++ /dev/null
@@ -1,30 +0,0 @@
------BEGIN CERTIFICATE-----
-MIIFDjCCAvagAwIBAgIBAzANBgkqhkiG9w0BAQsFADCBgTELMAkGA1UEBhMCUlUx
-DzANBgNVBAgMBk1vc2NvdzEPMA0GA1UEBwwGTW9zY293MRswGQYDVQQKDBJ0ZXN0
-LWxpYm1pY3JvaHR0cGQxITAfBgkqhkiG9w0BCQEWEm5vYm9keUBleGFtcGxlLm9y
-ZzEQMA4GA1UEAwwHdGVzdC1DQTAgFw0yMTA0MTAxMjM2MTZaGA8yMTIxMDMxNjEy
-MzYxNlowXDELMAkGA1UEBhMCUlUxDzANBgNVBAgMBk1vc2NvdzEPMA0GA1UEBwwG
-TW9zY293MRswGQYDVQQKDBJ0ZXN0LWxpYm1pY3JvaHR0cGQxDjAMBgNVBAMMBWhv
-c3QyMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAtSrwepZu2iycb2O+
-xTCUvSnljDwq7zlzaqyglWdT2fEPMft99OIoHy8P5nOXrYhDD82lGrbLyvw1WStf
-b25ZNAzezj6Pa2V2XVYyVU7B/lmjzumoOinZPACZdBI53d2kmWE+WsTOlDKjmvcT
-4eF/SEUMcmmAvp/sTEK2FE+mkMFN63C7B1jY1k8P5mjwTPOAL2/WU6I4H9HTDhPL
-f/bfIzFHqxKPwJJCRGXziEhz4ewNjkrPrXW9mm8c+SUmHispTxAm1JPnj3pmk7Yd
-Ju3XdIEjee+WH/WBt01u4gZpXq24cVWOB/FEziAKH6+ofgQVnHjdE/Uzu3PYLhwB
-eGcTOQIDAQABo4GyMIGvMAsGA1UdDwQEAwIFoDAMBgNVHRMBAf8EAjAAMBYGA1Ud
-JQEB/wQMMAoGCCsGAQUFBwMBMB0GA1UdDgQWBBR3q5u0JT2i/LzBiY/sq4HwPvZA
-ijAnBglghkgBhvhCAQ0EGhYYVGVzdCBsaWJtaWNyb2h0dHBkIGhvc3QyMBEGCWCG
-SAGG+EIBAQQEAwIGQDAfBgNVHSMEGDAWgBRYdUPApWoxw4U13Rqsjf9AHdbpLDAN
-BgkqhkiG9w0BAQsFAAOCAgEA1NY3DjvAQXG1QN6qcbSaGXKlvbblB1jMJbmDvbL2
-eQD9fGNhzzVDZxFbj8u9BmhGh8SbfcrRj7vmK8/lyqWgJxEAjIOTF6l4fanT72vi
-ucSz93LXcHTBzkTdvZyfshdkWtHEssTSQ4WPj/WFwIbCPFZ+My1GEU6+v7AjlzM5
-ob9dTIW0kWbnTU20TR54c9+rzS03obLXuYXoSq+y8AonhvdEAdeRiWx4siXCUeMo
-lFNn8Lr7yMgFKo21G3EMUuWmw+VvS1MAXNIKauHywOTrdDtllVWZxsPhP9IpLazf
-lQFn5JkYMaioeVscWfMEVgycX6c2A0SKKNVc+z9XhjGHAWMjHkVKWhqWNtB3oT7m
-hBKeXRWqOPZj2Fxl0Nc7cF/w13moKkPTQb3GDFTS0tFo7CrDcGEupgXfzNXl5ztE
-+09fh22D96PMtAYb6JTsXbI2xZ5fmLLAZHtIHz1KFdEmVG38CkvkzVtw6XDcEMRL
-mMuquyrw+3ug4oXDtT6mnlPr6MSEeP5bt4+sJhwyCYDmCkmF44K3rHm1cbaxsEtS
-P3jgYpjhgVBS75ZarpyZmj78usuDQwqPEGcW68RmdhvK89gajZKuqUcP0dl4TDPR
-CNGdTsd25O471k7ZEYkyaMWIIXjRzQwRQzMFsna/bewm79tpFI6Iji3mLH/o5jvA
-u6s=
------END CERTIFICATE-----
diff --git a/src/testcurl/https/host2.key b/src/testcurl/https/host2.key
deleted file mode 100644
index c3f16de3..00000000
--- a/src/testcurl/https/host2.key
+++ /dev/null
@@ -1,28 +0,0 @@
------BEGIN PRIVATE KEY-----
-MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQC1KvB6lm7aLJxv
-Y77FMJS9KeWMPCrvOXNqrKCVZ1PZ8Q8x+3304igfLw/mc5etiEMPzaUatsvK/DVZ
-K19vblk0DN7OPo9rZXZdVjJVTsH+WaPO6ag6Kdk8AJl0Ejnd3aSZYT5axM6UMqOa
-9xPh4X9IRQxyaYC+n+xMQrYUT6aQwU3rcLsHWNjWTw/maPBM84Avb9ZTojgf0dMO
-E8t/9t8jMUerEo/AkkJEZfOISHPh7A2OSs+tdb2abxz5JSYeKylPECbUk+ePemaT
-th0m7dd0gSN575Yf9YG3TW7iBmlerbhxVY4H8UTOIAofr6h+BBWceN0T9TO7c9gu
-HAF4ZxM5AgMBAAECggEAfwfRwRM0PTTkp9I31MqAsLbs5szVihItlJHgs4HVMM8v
-SWu/YPMnEN6N9pf8tcgCxdT7POUJtpXrzk90Ro0IbU/M9q61H6fwO/gRqPFlYzR1
-gMMjm0z2rnT5JIMXIjOkk4u1Di5Lb2ymfSVcQpk8B2+rD5pclM1MAhkQ1OMYtww0
-f61bzbKr4sZ5gXgi5FWPY31NYbX/N3sgp02/JGioJuIz2nNwLXNu76vkLMX/bNuN
-Lo0fCzw4CBy8anV2er3lN6355nkT9VPiJIdxXxMB5DMXhYT8jX6giROc7/+KlWX9
-agGHI+7CyCc2RgxED/9rEY80fBoDHwHjRBKORe9QJQKBgQDsjakF+jo/zs2uXHpZ
-gPYwYjafOFrG7J/kZPWTgwS6+RA7A+ilzhKx3o4uDVVrDdYmlnRpgsu/czwkyVVi
-O/q6r65D4KndfBD7Yv1xwBaSZhj2VwozCNSXN/2TMBI/jlX8VkJKjFsXesKiOAYS
-Bzfa5Y9xz3gaTlL86oqTFoCgEwKBgQDED6vSeI1HADHBIstOdLlo0gFLkYT+3ZRT
-LgEY2fTDFTgxRFWMGxzF4MKnJ3OcJucfgbc/spIKvdUL7GSjDZQzprwhzSExzLqe
-H3LCwIk3Rzxii/ou0Nm97RX8zLVK2YPKY7yoTXF+hD0NoHxKQhPcF7z9ospOKRAb
-293Z2ophAwKBgQCgSCQ8zp3lWkT3XeXw+3bUtdd47wBjRTXZ1aaoOuBbqCMdObrL
-jeQQ7VG9Vn+IOCngn5VlYKCZqoUYD+tw0l9DW4Jir7mpP0VqcFp2Vsd0jwVe9P8O
-MW+Kvv9sWegjdFhC0Fe+l3epYx66N+e1E3r/MQiO/xrMCEIm6OzYWSjyTQKBgFIE
-nER32eIR2r6I57PycH3BHHmYbqKP3zTM0firdREHF3LEu2TZErhUxhlXD64LODSd
-9yLWePG1aCKMyazMSWsyHRGm1nQWyRrM3R1LnhtZ2ZEALbDX/b7OJ5yRpmWhiNhp
-0Mo3W3sppsDEmjoWWmiAJbeVGjyIQQzmdXBlU+4xAoGAE3Byb3HkcNmPcV/vANhE
-snsQMLqoVzBNgaLfnuYUgIOORBfCMXgrvojA+aiSZbQKgdEXl71avT/1Ny/VwVoc
-ODS+KMxfCNlGyQ2KuIxzWewxyBzdiXBFtQ7ArFwtmbY5kTuuglVzHbhCi9oE8foH
-QW5tpN+3bzftqWO3urbizGY=
------END PRIVATE KEY-----
diff --git a/src/testcurl/https/mhdhost1.crt b/src/testcurl/https/mhdhost1.crt
new file mode 100644
index 00000000..775620ee
--- /dev/null
+++ b/src/testcurl/https/mhdhost1.crt
@@ -0,0 +1,30 @@
+-----BEGIN CERTIFICATE-----
+MIIFJjCCAw6gAwIBAgIBBTANBgkqhkiG9w0BAQsFADCBgTELMAkGA1UEBhMCUlUx
+DzANBgNVBAgMBk1vc2NvdzEPMA0GA1UEBwwGTW9zY293MRswGQYDVQQKDBJ0ZXN0
+LWxpYm1pY3JvaHR0cGQxITAfBgkqhkiG9w0BCQEWEm5vYm9keUBleGFtcGxlLm9y
+ZzEQMA4GA1UEAwwHdGVzdC1DQTAgFw0yMjA0MjAxODQ1MDVaGA8yMTIyMDMyNjE4
+NDUwNVowXzELMAkGA1UEBhMCUlUxDzANBgNVBAgMBk1vc2NvdzEPMA0GA1UEBwwG
+TW9zY293MRswGQYDVQQKDBJ0ZXN0LWxpYm1pY3JvaHR0cGQxETAPBgNVBAMMCG1o
+ZGhvc3QxMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAxb5LkwTfvNtW
+eFSYAPoEzqwVARWxDDR2PfQmmBtrHhMNEQyUCTg+Na00hEQbcZbz7fgS0FOY1hjy
+MFJamNGP031HvMS7zHNQJ4HbEz0d1EuePgjEvWaFMFncTKGn07Cn9e3Rcv2ihJ/I
+I8wf3ph/k/UTOv62YhIs2fMQM5LD6oX9ulKJhAaOvFT6hyrB1xe3nVhPT0PsrUXl
+ky253k7XXEAIWO6dLZnK377UiRDJInFS9FN/hojFb+8gcByo/n39LHMBtp5WGM5+
+xGwEHyYkwtnaM8IEKofbUNERQT3cgPv4zN7ny3LwljR2A1c5gHXmKO7G245tbfqb
+VtOh3+1bJQIDAQABo4HHMIHEMAsGA1UdDwQEAwIFoDAMBgNVHRMBAf8EAjAAMBYG
+A1UdJQEB/wQMMAoGCCsGAQUFBwMBMBMGA1UdEQQMMAqCCG1oZGhvc3QxMB0GA1Ud
+DgQWBBTTxzQ/fcuT83JlCzPx8qJzuHCG4jAnBglghkgBhvhCAQ0EGhYYVGVzdCBs
+aWJtaWNyb2h0dHBkIGhvc3QxMBEGCWCGSAGG+EIBAQQEAwIGQDAfBgNVHSMEGDAW
+gBRYdUPApWoxw4U13Rqsjf9AHdbpLDANBgkqhkiG9w0BAQsFAAOCAgEAMdHjWHxt
+51LH3V8Um3djUh+QYVnYhxFVX4CIXMq2DAg7iYw6tg2S1SOm3R+qEB/rAHxRw/cd
+nekjx0zm2rAnn4SsIQg5rXzniM4dJ2UQJrn7m88fb/czjXCrvf6hGIXCSAw/YyU0
+IkxXlOYv1mYNZBuVvFREUbqqK2Fn+ooIi816Q2UCQFMd+QI8+mn8eqK8XeJElKkt
+SR0WCLW7UkwYxOk73FCdrr1TlX9hAF76gTAK71OJN3F35hzg+pTAEe3nwFMZbG/k
+xHmVob51L+Op4Y15j1HoJxW0Ox+PuxVcp9EpC7Qb8UmtExAsYOuN2PP8hU+pmtE4
+iYpCDckx6kT66ssjndaKziRmoxX/czvKCAwVDder6Ofl4SwGUEpE+0oO2ux6iNrl
+iKxRCvyBqErFlzlEj7oO3agqe83jtTH8bXzOCtx36lNVjCuADYI4o7r5NVHaGIx6
+1q5BiZ5pj22EYgBZbRAgE1ZPxVBIZJKoHewmQqpcR5WsSeBIf8XDuytnadhv8NDN
+tpLMDry8KkUh7rtjrQIGb66BqCt0n1tz1/6nLlL9OF5kUt4902wCtN3SbotssfV3
+NCX9X4udquRdcFwNLCSfZ505FqhYbeSLraDHJ46/96V//Vx4oAnULdcCm4QgGQmP
+EGEJDJkaCKKTGMHAvZAtgRchc/RTf59jhoc=
+-----END CERTIFICATE-----
diff --git a/src/testcurl/https/mhdhost1.key b/src/testcurl/https/mhdhost1.key
new file mode 100644
index 00000000..8cc2dc41
--- /dev/null
+++ b/src/testcurl/https/mhdhost1.key
@@ -0,0 +1,28 @@
+-----BEGIN PRIVATE KEY-----
+MIIEvgIBADANBgkqhkiG9w0BAQEFAASCBKgwggSkAgEAAoIBAQDFvkuTBN+821Z4
+VJgA+gTOrBUBFbEMNHY99CaYG2seEw0RDJQJOD41rTSERBtxlvPt+BLQU5jWGPIw
+UlqY0Y/TfUe8xLvMc1AngdsTPR3US54+CMS9ZoUwWdxMoafTsKf17dFy/aKEn8gj
+zB/emH+T9RM6/rZiEizZ8xAzksPqhf26UomEBo68VPqHKsHXF7edWE9PQ+ytReWT
+LbneTtdcQAhY7p0tmcrfvtSJEMkicVL0U3+GiMVv7yBwHKj+ff0scwG2nlYYzn7E
+bAQfJiTC2dozwgQqh9tQ0RFBPdyA+/jM3ufLcvCWNHYDVzmAdeYo7sbbjm1t+ptW
+06Hf7VslAgMBAAECggEALshrttezW0oFNijFYY3FL2Q0//Gy1nFe/B9UNi5edFoL
+gFoad+fvh+F3iEdYutH82fMT+GeexCBYxCfnuTnzLhT4sOdWivNJJl+phe6yrPRK
+9uA6M5kar6rC3Ppt6z5jLmLaZ7ssBPaMcjOr4ozvugCEUTPL0H3+UH4Z+imh4kzw
+MgLsF8QkWPCF/fxsVhCwC3y3Dctl3oLh6qQFudYKtymRjk2I7zFAuQGsGoYonPbF
+1LZ6xHOjegXtqhC0pK8Kpn4N7LgYVVNPT1TAckQ9GDDi0IgMYwIsSQb/ncaV5Cwb
+9tPqBv1Oi2ayINeXrG5J/G8S1cEbIzdSL8ZqG+CcjQKBgQD1c89Rh/Hbc0zxxgdR
+eGx6oOtNjAZ3zgnNyVnXiZ7mXtbCcjMHJhcW41RL8lg+mhKfj9MgXamDP/R4X2In
+H358dEekfN4wjogeXs9lmh48cGC7zez4/RS752OmrwmhOqkKDa7dy+SLR5QJ3L2n
+4k/2eQwTlXpuPIMrBT1hUvsA3wKBgQDOPaTy03oXX1bwwX6sHUTrQTKAC52v4hQl
+bya+6t+VGpj+cdWzxRFoR5mN85b9Kkj1nAFx+6zioOPLUOZhnnqihlQBwm06gH1p
+v3qkxuXOhnMPJsNg4lltUq/18UlSQwFZuKr7G2vDhckHwZdlu06P/rlA9K5Kxjzi
+gX3I+XoQewKBgQDrn/YYXXmW4jOuMR0bX5A7lDjuY4pd/iO5Mh6V453vpoFhfoFv
+zmgB588nbQi7Z+qS1F2nx2IQBhgoaeBukDQ7QuD3jYs6b8lJ5lgQQAfgmzyxbPic
++U6rJ3CpNYT4Crj1VrdUYgQOlHMPmKFUBdQfVop6TleOdXaxmMEYqbEdXwKBgQCz
+xZwQZjJYSSyZc7CdCm5Wul/wqS9sbp6s+rRFWqpFaAfQUx26M582zKKWz6vfRYqP
+PMsttfk/Gos1YHFQyjmPjZOQbQ+VHQc0tEmNdCpA2YVVwa4wt1zIJHlo4kfNQsbc
+lFHFzGMk7WsMLb1wWdLjRV/ptN5wI1hTABjKpFu4HQKBgExQNTHMxzpAAh9XFiJ3
+oGvf+9+JtjmrgKsR0T7q60xjOYH0E2rznvvBVNeZlAc+tjawmedxiuiitmjQUteG
+hW4ncX0UcqFPm7Ahqo9+NDemJlV7DYHRcvL/xVNUbr8yV7OpHILd4Hk/s8J4QGl/
+bC1zEehy8q0jMywvSR8vsS1v
+-----END PRIVATE KEY-----
diff --git a/src/testcurl/https/mhdhost2.crt b/src/testcurl/https/mhdhost2.crt
new file mode 100644
index 00000000..c32fb6e4
--- /dev/null
+++ b/src/testcurl/https/mhdhost2.crt
@@ -0,0 +1,30 @@
+-----BEGIN CERTIFICATE-----
+MIIFJjCCAw6gAwIBAgIBBjANBgkqhkiG9w0BAQsFADCBgTELMAkGA1UEBhMCUlUx
+DzANBgNVBAgMBk1vc2NvdzEPMA0GA1UEBwwGTW9zY293MRswGQYDVQQKDBJ0ZXN0
+LWxpYm1pY3JvaHR0cGQxITAfBgkqhkiG9w0BCQEWEm5vYm9keUBleGFtcGxlLm9y
+ZzEQMA4GA1UEAwwHdGVzdC1DQTAgFw0yMjA0MjAxODQ1MDVaGA8yMTIyMDMyNjE4
+NDUwNVowXzELMAkGA1UEBhMCUlUxDzANBgNVBAgMBk1vc2NvdzEPMA0GA1UEBwwG
+TW9zY293MRswGQYDVQQKDBJ0ZXN0LWxpYm1pY3JvaHR0cGQxETAPBgNVBAMMCG1o
+ZGhvc3QyMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy2v1c39RQPgG
+EqhYisiQX5QDTJFCr93B/KC96AqxXfUqxYHQZHHByeh2yHX5XeY+cwiza9Pb8kB5
+Wyu5IV7pHYbLXetUQ7Ha8oCXcXZBihzMGjdTQApBvKES2rotkgxIArzFRk/KnO3D
+2r87E2/fHqlHjg0Y8d58i846m7GSFpH2uAYogm2Zhi2ojE4O/qfaO6cSTdjUhDbM
+iDkQlf/fkq9o3ZeGzKXdzLC5PeIJ/7HbEcBJ0e6lyAFbYZtpTAtQ6/tuqAZTs3tM
+G5Yq3cAZY7x4vGECZaAWZAlgstBpCYd8b/dlXa8qvx4JuMF4q+7vVbgwHVoXsAzm
+ZAxSNy+dmQIDAQABo4HHMIHEMAsGA1UdDwQEAwIFoDAMBgNVHRMBAf8EAjAAMBYG
+A1UdJQEB/wQMMAoGCCsGAQUFBwMBMBMGA1UdEQQMMAqCCG1oZGhvc3QyMB0GA1Ud
+DgQWBBSB0kA6mqq1/wMzKke35+L6aZym9zAnBglghkgBhvhCAQ0EGhYYVGVzdCBs
+aWJtaWNyb2h0dHBkIGhvc3QyMBEGCWCGSAGG+EIBAQQEAwIGQDAfBgNVHSMEGDAW
+gBRYdUPApWoxw4U13Rqsjf9AHdbpLDANBgkqhkiG9w0BAQsFAAOCAgEAhEeSsKYb
+KWVhC3bVxXX4EynaHVWwrQaqNMtogt2akKBfZRzM6CHVvts8IrvcMdS4ZvXB2kL8
+wDsYpVsNVgLLuSKXH2xJPZmYrGsUu++UcR4umvuV1kwFcA/Dc5XkW1EMcMvSPDs2
+/gJFRCpx3UAs2/KCL2KASHGiOcNSf8aRLxI6KqWqQLxsZdK02HUNodqiG2UXIaHg
+shD7rAMSDN4CdG4cdSX3l45aL+BoYcEKi5XUoJMRjNrAGKMWUku089hr1gI617Jx
+gHKZxvSizGxCHhMkAPTT+9UyNFzd+3dS6cHx8oTFpnamNgMin3C88GCmu5YN6gCi
+af3imb8OaHFJ0k9M9acM++5sRfvRgZ/tQWnPdyg5TlLwwF14yLbz6e/DbxlmP7PG
+3W3FbVST/creb2KTtwWFu187uKl4v1Xe5DC0I2gcOzq2g0GHbu8WjuD45Os/lc4v
+cqGctYRMUfqshmsK9PQCZmPcC5nqXf6iO8L1P9Gi0VDwT/WeHUvi87UVYsUgVuVt
+0RxCE5Nqad8S7ziQj9Aa3xspPf7CKw6Obbp/UNui/YwaDufJho7qsBpp1lMnG5f4
+qA8+yKUVADTnV5tJqgcDiRyBxOzb2CaGezTqNshUkVIpkHZsCpSdu9QrEOIzcaaA
+WXmQOPOpxW5QVf22mPJh/GZaqZ3/ZMmFabI=
+-----END CERTIFICATE-----
diff --git a/src/testcurl/https/mhdhost2.key b/src/testcurl/https/mhdhost2.key
new file mode 100644
index 00000000..7e870318
--- /dev/null
+++ b/src/testcurl/https/mhdhost2.key
@@ -0,0 +1,28 @@
+-----BEGIN PRIVATE KEY-----
+MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQDLa/Vzf1FA+AYS
+qFiKyJBflANMkUKv3cH8oL3oCrFd9SrFgdBkccHJ6HbIdfld5j5zCLNr09vyQHlb
+K7khXukdhstd61RDsdrygJdxdkGKHMwaN1NACkG8oRLaui2SDEgCvMVGT8qc7cPa
+vzsTb98eqUeODRjx3nyLzjqbsZIWkfa4BiiCbZmGLaiMTg7+p9o7pxJN2NSENsyI
+ORCV/9+Sr2jdl4bMpd3MsLk94gn/sdsRwEnR7qXIAVthm2lMC1Dr+26oBlOze0wb
+lirdwBljvHi8YQJloBZkCWCy0GkJh3xv92Vdryq/Hgm4wXir7u9VuDAdWhewDOZk
+DFI3L52ZAgMBAAECggEBAIS8cXFoBpEqRmwuRXhp3ys+3dg8gRNY1JgQG1sqfwoc
+TEiMqHqicB1b/wZXVNycvOs7JjiaCc9NmuKO6UKJN/v4VQN354g0qfXLSwbSb3m7
+yMLijwQerT50rGTlT48ZTHPc0a1Lq54y17YJSncobKMJOpPKoBhTYVmovD2T5Qus
+EDFg8wLBA79YZnrImbt9hcRsi/bM341V3aSoPFdHz6NTi+5WA08oTiGCv6gThV8T
+PUCiy1uDJZ1gk3xne+APZfdTutVtwCYVw7mhgTHiL5tL1KCjDjzK3XgnfAWaN+wG
+ex9EYbn/uAxN57UrvdyvGtuBo01vs/XtsVlpRpmlBnECgYEA+L8D8UY36IubeMAl
+xFCoFSXSmT+2YtA6ISg4/8a14wpwAskJPW/Nf8CzzwbGA7QaDVeK4rTGDOwNQJV8
+/fiTaG6vDI7GS3UkvSKStfJzoMRBTzCYp/TbkG/jW2i8tR1AATlQlGllx0BQVXhC
+iYpR/WzbxGi5iyB4K6WPbSKboh8CgYEA0VqUR7w+2PEEFXyFRL2drpzkWKk5RiIa
+cIzQk3VlrHztJp0g6w8MiWCIpzL+7BuBy72byxJwpFmOz4HYsEqc6mOF4xXezfYh
+oN0P1+/aH4NrpWmjb65TGrtwrmi2sJrmATmr9WKFxEIpgrbotOTDF7pIx9pY2ScA
+aohOxqb7eUcCgYAHtHL80D3/GAPy05DX6d+q+Abz9ENEAEssp8BMO+16YOJjU7LT
+klj9MgzfxsfvaW69Jw8IQq03zUAD1h2PCFoYjAUkEHAX+kLvENkWhbILMskLGOhB
+m5YJfU2/kRj3SzamUw4p6rHaYCWc4CK/e+daQDr2dH/6zUCrqW8t5DqJ5QKBgF8E
+voIkhV3PXiwmXRJLAXNMEDPRcoZLWja1IsGaqe/0r2o0LMmjBeygHMXOVndxMKL5
+RumPUAK4ByJVa7Tv2HJlg1IDDiHq0W6ChvtaCGT/L+9el+hLdbqPUmBGdIyJcVUj
+CNIRyma+JLsIK2xW29k8GmZiyqqckgrIHQD6ru5nAoGAAcdAHQojvGs5dQ2Agqcn
+EOgF1dic7VgCjAEG4CNCydzskE92jEOgv2DI+oBPoLrtktHbewFrsxBBNbS+PcHY
+DyV5/WXf8wwwdkG1VLyw+dPeWNGWanvjllplckDXSGz1ia0s7ENNAt8o2MXdlZrQ
+MMHn0wo/1rTa1YroHF8cKVA=
+-----END PRIVATE KEY-----
diff --git a/src/testcurl/https/test_https_sni.c b/src/testcurl/https/test_https_sni.c
index 1c813c22..a1272a18 100644
--- a/src/testcurl/https/test_https_sni.c
+++ b/src/testcurl/https/test_https_sni.c
@@ -214,9 +214,9 @@ do_get (const char *url, int port)
   /* TODO merge into send_curl_req */
   curl_easy_setopt (c, CURLOPT_SSL_VERIFYPEER, 0L);
   curl_easy_setopt (c, CURLOPT_SSL_VERIFYHOST, 2L);
-  sprintf (buf, "host1:%d:127.0.0.1", port);
+  sprintf (buf, "mhdhost1:%d:127.0.0.1", port);
   dns_info = curl_slist_append (NULL, buf);
-  sprintf (buf, "host2:%d:127.0.0.1", port);
+  sprintf (buf, "mhdhost2:%d:127.0.0.1", port);
   dns_info = curl_slist_append (dns_info, buf);
   curl_easy_setopt (c, CURLOPT_RESOLVE, dns_info);
   curl_easy_setopt (c, CURLOPT_FAILONERROR, 1L);
@@ -277,8 +277,10 @@ main (int argc, char *const *argv)
     return 77;
   }
 
-  load_keys ("host1", ABS_SRCDIR "/host1.crt", ABS_SRCDIR "/host1.key");
-  load_keys ("host2", ABS_SRCDIR "/host2.crt", ABS_SRCDIR "/host2.key");
+  load_keys ("mhdhost1", ABS_SRCDIR "/mhdhost1.crt",
+             ABS_SRCDIR "/mhdhost1.key");
+  load_keys ("mhdhost2", ABS_SRCDIR "/mhdhost2.crt",
+             ABS_SRCDIR "/mhdhost2.key");
   d = MHD_start_daemon (MHD_USE_THREAD_PER_CONNECTION
                         | MHD_USE_INTERNAL_POLLING_THREAD | MHD_USE_TLS
                         | MHD_USE_ERROR_LOG,
@@ -302,9 +304,9 @@ main (int argc, char *const *argv)
     }
     port = (int) dinfo->port;
   }
-  if (0 != do_get ("https://host1/", port))
+  if (0 != do_get ("https://mhdhost1/", port))
     error_count++;
-  if (0 != do_get ("https://host2/", port))
+  if (0 != do_get ("https://mhdhost2/", port))
     error_count++;
 
   MHD_stop_daemon (d);
-- 
GitLab

